### DISCUSSION TITLE
SpatieMediaLibraryFileUpload not uploading to specified directory and visibility

### DISCUSSION THREAD

**Discussion by joshembling** [filamentphp/filament#6643] [General]
I am successfully uploading to an S3 bucket, but not to my specific sub directory. I am then getting a 403 when trying to access the image (I'm assuming because visibility isn't set). 

Here is the snippet: 
```
SpatieMediaLibraryFileUpload::make('image')
    ->image()
    ->disk('s3')
    ->visibility('public')
    ->directory('checklists')
```

As above I am setting visibilty to 'public' and the directory to 'checklists'.

filesystems config looks like this:
```
's3' => [
    'driver' => 's3',
    'key' => env('AWS_ACCESS_KEY_ID'),
    'secret' => env('AWS_SECRET_ACCESS_KEY'),
    'token' => env('AWS_SESSION_TOKEN'),
    'region' => env('AWS_DEFAULT_REGION'),
    'bucket' => env('AWS_BUCKET'),
    'url' => 'https://' .env('AWS_BUCKET') . '.s3-' . env('AWS_DEFAULT_REGION') . '.amazonaws.com/',
    'endpoint' => env('AWS_ENDPOINT'),
    'use_path_style_endpoint' => env('AWS_USE_PATH_STYLE_ENDPOINT', false),
    'throw' => false,
],
```

Is there an issue with how I'm chaining methods on `SpatieMediaLibraryFileUpload`? Anything obvious I'm missing that is needed to make this work?

**Comment by danharrin** [comment]
Spatie media library itself uses its own directory structure, sorry about that. It's not really something we control.

**Comment by elmudometal** [comment]
to change the directory in Media library you must do a CustomPathGenerator, but you must also take into account that when removing "Media library" it removes everything in that folder and not by file
